### PR TITLE
chore: bundle linux binaries

### PR DESCRIPTION
**Pull request by EthnTuttle**

### CODE REVIEW

**Comment by EthnTuttle** [comment]
#2768
Just waiting to see what happens when it runs

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -332,3 +330,39 @@ jobs:
         webhook: ${{ secrets.DISCORD_WEBHOOK }}
         # current job is a success, but that's not what we're interested in
         status: failure
+
+  deb-binary:
```
Comment (dpc) [inline .github/workflows/ci-nix.yml:334]: We should probably do this in `Release packages` workflow, as both will be building release builds of stuff, so possibly can re-use most of the work.

Comment (EthnTuttle) [reply .github/workflows/ci-nix.yml:340]: I've added the `Release packages` to the artifact upload.
Is that what you meant?

File: .github/workflows/ci-nix.yml [hunk 2]
```diff
@@ -290,7 +288,7 @@ jobs:
           - pkg: gateway-pkgs
             bin: gateway-cli
 
-    runs-on: buildjet-4vcpu-ubuntu-2004
+    runs-on: ubuntu-20.04
```
Comment (EthnTuttle) [inline .github/workflows/ci-nix.yml:291]: I will change this before marking this PR ready, but I don't have this image in my repo's CI

File: .github/workflows/ci-nix.yml (hunk 1 above)
Comment (dpc) [reply .github/workflows/ci-nix.yml:340]: Nope.

There is already a job "Release packages" https://github.com/fedimint/fedimint/blob/8864e239e539c716b4f970397f508598dbb1c383/.github/workflows/ci-nix.yml#L281 and I think under the hood is already has to build the same thing (release versions of packages). Bundling into deb/rpm is just an extra step.

So by doing all the stuff here in that job, we could avoid doing the same thing twice.

File: .github/workflows/ci-nix.yml [hunk 3]
```diff
@@ -332,3 +336,39 @@ jobs:
...
         # current job is a success, but that's not what we're interested in
         status: failure
+
+  deb-binary:
+    name: "Bundling for DEB and RPM"
+    runs-on: ubuntu-20.04 
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+        if: github.ref == 'refs/heads/master'
```
Comment (elsirion) [inline .github/workflows/ci-nix.yml:346]: The whole job should be conditional. Making single steps conditional makes later steps fail right now and adding the same if condition to all of them to fix that seems suboptimal.

**Comment by dpc** [comment]
Took a stab at wrapping it up in:  #2859

**Comment by justinmoon** [comment]
https://github.com/fedimint/fedimint/pull/2859 is merged, closing this

**Pull request closed without merging**